### PR TITLE
Add Wi-Fi example

### PR DESCRIPTION
**Pull request by SergioGasquez**
Initial version of the Wi-Fi exercise. Happy to modify/change anything!

### CODE REVIEW

**Comment by bjoernQ** [comment]
Thanks for working on this!

I guess the example / exercise is fine - it just feels a bit weird going from basic interrupt handling to making an http request via WiFi.

Maybe we need some more introduction like explaining that we have support for WiFi via an additional crate and that there is embbeded-svc to control it and how it relates to smoltcp ... something like that. I don't have an idea how it should really look like - and maybe it's a "me" thing

But yeah, looking at the std training they even start with http after "Hello World" 🤔 so would be definitely good to hear from others what they think about it. Also, this is the initial version - so maybe just something to refine later

**Comment by SergioGasquez** [comment]
I think you are completely right, adding those kinds of introduction should help the reader a lot, specially if it's newbie. I just added a small intro, let me know what do you think about it. Im a happy to rewrite/expand/change it as needed

File: book/src/03_5_http_client.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+Before jumping to the exercise, let's explore how Wi-Fi works in `no_std` Rust for Espressif devices.
+
+## Wi-Fi Ecosystem
+
+Wi-Fi support comes in the [`esp-wifi` crate][esp-wifi]. The `esp-wifi` is home to the Wi-Fi, Bluetooth and ESP-NOW driver implementations for `no_std` Rust.
+Check the repository README for current support, limitations and usage details.
+
+There are some other relevant crates, on which `esp-wifi` depends on:
+- [`embedded-svc`][embedded-svc]: Contains traits for features such as wifi, networking, HTTPD, and logging.
```
Comment (bjoernQ) [inline book/src/03_5_http_client.md:12]: While this is generally true we currently only have a bare-metal implementation for the `wifi` module (living in esp-wifi) - so maybe just mention that?

off-topic: I thought about implementing some other traits (in other crates) for bare-metal. I don't agree with all of them in general and the for some parts with a few details t.b.h. - but that discussion doesn't belong here - will raise that question in our next community meeting (if I don't forget about it)

**Review by bjoernQ — APPROVED** [review]
I think this is a great start!

Only one comment regarding embedded-svc ... what is written here is technically correct but maybe we shouldn't mention it 😄

File: book/src/03_5_http_client.md (hunk 1 above)
Comment (SergioGasquez) [reply book/src/03_5_http_client.md]: Just updated this information in https://github.com/esp-rs/no_std-training/pull/7/commits/88bbba996de35689d2dfbf3c3036f971d4d2be45

**Pull request merged**